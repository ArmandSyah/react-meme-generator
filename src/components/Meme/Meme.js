import React, { useState } from "react"
import memesData from "../../data/memesData"

export default function Meme() {
    const memes = memesData['data']['memes']

    const [memeImage, setMemeImage] = useState('');
    
    const handleClick = () => {
        const randomIndex = Math.floor(Math.random()*memes.length);
        const randomMeme = memes[randomIndex];
        
        setMemeImage(randomMeme.url);

        console.log(randomMeme.url)
    }

    return (
        <main>
            <div data-testid="form" className="form">
                <input 
                    type="text"
                    placeholder="Top text"
                    className="form--input"
                />
                <input 
                    type="text"
                    placeholder="Bottom text"
                    className="form--input"
                />
                <button
                    data-testid="meme-button"
                    onClick={handleClick} 
                    className="form--button"
                >
                    Get a new meme image ðŸ–¼
                </button>
                
            </div>
            <img src={memeImage} className="meme--image"/>
        </main>
    )
}