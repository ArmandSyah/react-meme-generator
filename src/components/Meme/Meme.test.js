import React from "react";
import { screen, render, cleanup, fireEvent } from "@testing-library/react";

import Meme from './Meme';

afterEach(cleanup);

it('renders properly', async () => {
    render(<Meme />)

    const formElement = screen.getByTestId("form");
    expect(formElement).toBeVisible();

    // Image should be empty at start
    const memeImage = screen.getByRole('img')
    expect(memeImage).toHaveAttribute('src', '')

    const buttonElement = screen.getByTestId("meme-button")
    expect(buttonElement).toHaveClass("form--button")
    
    fireEvent.click(buttonElement);
    expect(memeImage).toHaveAttribute('src', expect.stringMatching(/jpg|png/i));
});

it('renders', async () => {
    const { asFragment } = render(<Meme />)
    expect(asFragment()).toMatchSnapshot();
});