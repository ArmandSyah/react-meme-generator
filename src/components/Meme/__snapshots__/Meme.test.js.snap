// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <main>
    <div
      class="form"
      data-testid="form"
    >
      <input
        class="form--input"
        placeholder="Top text"
        type="text"
      />
      <input
        class="form--input"
        placeholder="Bottom text"
        type="text"
      />
      <button
        class="form--button"
        data-testid="meme-button"
      >
        Get a new meme image ðŸ–¼
      </button>
    </div>
    <img
      class="meme--image"
      src=""
    />
  </main>
</DocumentFragment>
`;
